libs/environment/src/components/network-loader/network-loader.tsx

```diff
@@ -0,0 +1,163 @@
+import { useState, useEffect, useMemo } from 'react';
+import type { ReactNode, FC } from 'react';
+import type { ApolloClient } from '@apollo/client';
+import { ApolloProvider } from '@apollo/client';
+import {
+  Callout,
+  Intent,
+  Button,
+  Icon,
+  Loader,
+} from '@vegaprotocol/ui-toolkit';
+import { t } from '@vegaprotocol/react-helpers';
+import { useEnvironment } from '../../hooks';
+import type { ConfigStatus } from '../../types';
+
+type NetworkLoaderProps<T> = {
+  children?: ReactNode;
+  skeleton?: ReactNode;
+  createClient: (url: string) => ApolloClient<T>;
+};
+
+type StatusComponentProps = {
+  status: ConfigStatus;
+  children?: ReactNode;
+};
+
+type MessageComponentProps = {
+  children: ReactNode;
+};
+
+type ErrorComponentProps = MessageComponentProps & {
+  showTryAgain?: boolean;
+};
+
+const Error: FC<ErrorComponentProps> = ({
```

Comment:
chore (blocking): Using FC<> here is pretty redundant IMO, our general code style here is to just type props directly and not use ComponentTypes (like you already have on line 38)

```suggestion
const Error = ({
```

Reply:
true, using both is stupid here. FC does inject the children type for convenience, so i don't have to manually append it to the types i'm using, but happy to add it manually, if that's the preferred way.

Reply:
Up for discussion, but lets keep to the convention for now. I prefer manually typing children as then you know it should be used within